File: 50/main.c
#include <math.h>
#include <stdio.h>

int is_prime(int number) 
{
  int i;
  int flag = 1;

  for (i = 2; i <= sqrt(number); i++) {
    if (number % i == 0) {
      flag = 0;
      break;
   }
  }

  if ( number <= 1 )
    flag = 0;

  if (flag == 1 )
    return 1;
  else
    return 0;
}

int main() 
{
  char str[1000] = "2";
  int sum = 2, count = 2;

  while (sum<1000000)
  {
    count=count+1;
    if (is_prime(count))
    {
      sum=sum+count;
      snprintf(str, 1000, "%s+%d", str, count);
      printf("%s=%d\n",str,sum);
      if (is_prime(sum))
      {
        printf("Sum %d is prime.\n",sum);
      }
    }
  }
  printf("Result is %d.\n",sum);
}

File: 50/main.py
#!/usr/bin/env python3
"""
Module Docstring
"""
import math

MAX = 100

def is_prime(n):
  for i in range(2,int(math.sqrt(n))+1):
    if (n%i) == 0:
      return False
  return True

def reverse(num):
  reversed_num = 0
  while num != 0:
    digit = num % 10
    reversed_num = reversed_num * 10 + digit
    num //= 10
  return reversed_num

def main():

  ###
  # Generate the prime list first
  cnt = 0
  x = 3
  prime_list = [2]
  while x < MAX:
    if is_prime(x):
      cnt = cnt + 1
      prime_list.append(x)    
    x = x + 2
  
  ###
  # Get the lenght
  list_len=len(prime_list)
  list_sum=sum(prime_list)
    
  ###
  # Count max
  cnt_max = 0
  num_max = 0

  cnt_a = 0
  cnt = 0
  while cnt_a < list_len-1:
    cnt_b = cnt_a
    tmp_sum = 0
    cnt = 0
    while cnt_b < list_len-1:
      tmp_sum = tmp_sum + prime_list[cnt_b]
      if is_prime(tmp_sum):
        cnt = cnt + 1
        if cnt_max < cnt:
          cnt_max = cnt
          num_max = tmp_sum
      cnt_b = cnt_b + 1
    cnt_a = cnt_a + 1
  print(cnt_max, num_max)
  

if __name__ == "__main__":
  main()
